server.test.js

```diff
@@ -0,0 +1,51 @@
+const { EventEmitter } = require('events')
+const { mkdtemp, rm, writeFile } = require('fs/promises')
+const { tmpdir } = require('os')
+const path = require('path')
+
+const createServer = require('./server')
+
+describe('server', () => {
+  let tmpd
+  let wsServerMock
+  let socketMock
+  let abortController
+
+  beforeEach(async () => {
+    tmpd = await mkdtemp(path.join(tmpdir(), 'next-remote-refresh_'))
+    wsServerMock = Object.assign(new EventEmitter(), {
+      close: jest.fn(),
+    })
+    socketMock = Object.assign(new EventEmitter(), {
+      send: jest.fn(),
+    })
+    jest.spyOn(require('http'), 'createServer').mockReturnValue(
+      Object.assign(new EventEmitter(), {
+        address: jest.fn().mockReturnValue({ port: 2000 }),
+        close: jest.fn(),
+        listen: jest.fn(),
+      }),
+    )
+    jest.spyOn(require('ws'), 'Server').mockReturnValue(wsServerMock)
+    abortController = new AbortController()
+  })
+
+  afterEach(async () => {
+    abortController.abort()
+    jest.restoreAllMocks()
+    await rm(tmpd, { recursive: true })
+  })
+
+  it('watches a directory for changes', async () => {
+    createServer({ paths: tmpd, signal: abortController.signal })
+    wsServerMock.emit('connection', socketMock)
+
+    // TODO: make createServer async
```

Comment:
Curious the need for `createServer` being async, is this more for making testing easier?

Reply:
Yes it would mostly be to assist in testing. A callback function would also work, ultimately unit tests just need a handle for when a) the websocket server is ready and b) chokidar's file watcher is ready